### PR TITLE
Bp annotation updates

### PR DESCRIPTION
**Pull request by hahn-kev**
This is a decent starting point for adding annotations to blueprints. This one just adds annotations for Unit blueprints, excluding the Weapons, Display, and Enhancements. Most of the comments here come directly from the wiki here: https://wiki.faforever.com/en/Blueprints.

I've also included some new files with empty classes and proper function annotations for anyone else to come along and define annotations for any of the other blueprint types (props, projectiles emitters etc...)

This is already a decent sized PR, so we can just merge as is, or wait until I do weapons and display.

### CODE REVIEW

**Comment by Ejsstiil** [comment]
Awesome, great work!

**Comment by Garanas** [comment]
Very exciting!

File: engine/Sim/Blueprints/BlueprintBaseDefinition.lua [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---@meta
+---@declare-global
+
+---@class BlueprintBase
+---@field BlueprintId? string -- BlueprintId is used for modding an existing blueprint. An example for usage is BlueprintId = 'uel0101' which would be the UEF Land Scout.
+---@field Merge? boolean -- Merge is used for moding an existing blueprint. With Merge = 'true' the BlueprintId must be set.
```
Comment (Garanas) [inline engine/Sim/Blueprints/BlueprintBaseDefinition.lua:6]: `moding` -> `modding`

**Comment by hahn-kev** [comment]
@Garanas this is good to go, I'll do weapons in a later PR

**Pull request merged**